libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -0,0 +1,507 @@
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/**
+ * @file transport_secure_sockets.c
+ * @brief Implementation of transport interface API with Secure Sockets.
+ */
+
+/* Standard includes. */
+#include <stdlib.h>
+#include <string.h>
+
+/* TCP/IP abstraction includes. */
+#include "iot_secure_sockets.h"
+#include "transport_secure_sockets.h"
+
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Set up the context's send and receive timeout.
+ *
+ * @param[in] tcpSocket The context to be set.
+ * @param[in] sendTimeoutMs Timeout for transport send.
+ * @param[in] recvTimeoutMs Timeout for transport recv.
+ *
+ * @return #SOCKETS_ERROR_NONE or error code from Secure Sockets.
+ */
+static int32_t transportTimeoutSetup( Socket_t tcpSocket,
+                                      uint32_t sendTimeoutMs,
+                                      uint32_t recvTimeoutMs );
+
+/**
+ * @brief Sets up a TLS session on top of a TCP connection using the Secure Sockets API.
+ *
+ * @param[out] pNetworkContext The output parameter to return the created network context.
+ * @param[in] pServerInfo Server connection info.
+ * @param[in] pSocketsConfig socket configs for the connection.
+ *
+ * @return #TRANSPORT_SOCKET_STATUS_SUCCESS on success;
+ *         #TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, #TRANSPORT_SOCKET_STATUS_INSUFFICIENT_MEMORY,
+ *         #TRANSPORT_SOCKET_STATUS_INVALID_CREDENTIALS, #TRANSPORT_SOCKET_STATUS_INTERNAL_ERROR,
+ *         #TRANSPORT_SOCKET_STATUS_DNS_FAILURE, #TRANSPORT_SOCKET_STATUS_CONNECT_FAILURE on failure.
+ */
+static TransportSocketStatus_t establishConnect( NetworkContext_t * pNetworkContext,
+                                                 const ServerInfo_t * pServerInfo,
+                                                 const SocketsConfig_t * pSocketsConfig );
+
+/**
+ * @brief Set up TLS configurations for the socket.
+ *
+ * @param[in] pSocketsConfig The session configuration.
+ * @param[in] tcpSocket An initialized socket to secure.
+ * @param[in] pHostName Remote server name for SNI.
+ * @param[in] hostnameLength The length of `pHostName`.
+ *
+ * @return #SOCKETS_ERROR_NONE or #SOCKETS_SOCKET_ERROR.
+ */
+static int32_t tlsSetup( const SocketsConfig_t * pSocketsConfig,
+                         Socket_t tcpSocket,
+                         const uint8_t * pHostName,
+                         size_t hostnameLength );
+
+
+/*-----------------------------------------------------------*/
+
+int32_t SecureSocketsTransport_Send( NetworkContext_t * pNetworkContext,
+                                     const void * pMessage,
+                                     size_t bytesToSend )
+{
+    size_t bytesSent = 0;
+    int32_t TransportSocketStatus = SOCKETS_ERROR_NONE;
+    Socket_t xSocket = SOCKETS_INVALID_SOCKET;
```

Comment:
`xSocket` is a FreeRTOS kernel style naming convention. As the rest of the naming in the file doesn't follow the FreeRTOS style, it would be fine to just use `socket` to be consistent with a single style through the file

Reply:
removed all xSocket in b719c48142e222a8b58ca00e4d06027d03c6e5da